src/dc_receive_imf.rs

```diff
@@ -60,673 +59,613 @@ pub unsafe fn dc_receive_imf(
-                    dc_add_or_lookup_contacts_by_mailbox_list(
-                        context,
-                        (*fld_from).frm_mb_list,
-                        0x10i32,
-                        from_list,
-                        &mut check_self,
-                    );
-                    if 0 != check_self {
-                        incoming = 0i32;
-                        if 0 != dc_mimeparser_sender_equals_recipient(&mime_parser) {
-                            from_id = 1i32 as uint32_t
-                        }
-                    } else if dc_array_get_cnt(from_list) >= 1 {
-                        from_id = dc_array_get_id(from_list, 0i32 as size_t);
-                        incoming_origin =
-                            dc_get_contact_origin(context, from_id, &mut from_id_blocked)
+        }
+        field = dc_mimeparser_lookup_field(&mime_parser, "From");
+        if !field.is_null() && (*field).fld_type == MAILIMF_FIELD_FROM as libc::c_int {
+            let fld_from: *mut mailimf_from = (*field).fld_data.fld_from;
+            if !fld_from.is_null() {
+                let mut check_self: libc::c_int = 0;
+                let from_list: *mut dc_array_t = dc_array_new(16i32 as size_t);
+                dc_add_or_lookup_contacts_by_mailbox_list(
+                    context,
+                    (*fld_from).frm_mb_list,
+                    0x10i32,
+                    from_list,
+                    &mut check_self,
+                );
+                if 0 != check_self {
+                    incoming = 0i32;
+                    if 0 != dc_mimeparser_sender_equals_recipient(&mime_parser) {
+                        from_id = 1i32 as uint32_t
                     }
-                    dc_array_unref(from_list);
+                } else if dc_array_get_cnt(from_list) >= 1 {
+                    from_id = dc_array_get_id(from_list, 0i32 as size_t);
+                    incoming_origin = dc_get_contact_origin(context, from_id, &mut from_id_blocked)
                 }
+                dc_array_unref(from_list);
             }
-            field = dc_mimeparser_lookup_field(
-                &mime_parser,
-                b"To\x00" as *const u8 as *const libc::c_char,
-            );
-            if !field.is_null() && (*field).fld_type == MAILIMF_FIELD_TO as libc::c_int {
-                let fld_to: *mut mailimf_to = (*field).fld_data.fld_to;
-                if !fld_to.is_null() {
+        }
+        field = dc_mimeparser_lookup_field(&mime_parser, "To");
+        if !field.is_null() && (*field).fld_type == MAILIMF_FIELD_TO as libc::c_int {
+            let fld_to: *mut mailimf_to = (*field).fld_data.fld_to;
+            if !fld_to.is_null() {
+                dc_add_or_lookup_contacts_by_address_list(
+                    context,
+                    (*fld_to).to_addr_list,
+                    if 0 == incoming {
+                        0x4000i32
```

Comment:
Are there no constants for these magic numbers somewhere?  I've found that the c-to-rust translation ate a lot of constants that need to be put back.

Reply:
afaik, not all constants are yet re-defined.

(there were constants in the c-core, `DC_ORIGIN_INCOMING_TO` in this case which was defined at [dc_contact.h](https://github.com/deltachat/deltachat-core/blob/master/src/dc_contact.h#L42) and also used in [dc_receive_imf.c](https://github.com/deltachat/deltachat-core/blob/master/src/dc_receive_imf.c#L1145). during the automatic c-to-rust conversion these information were lost as the conversion called the preprocessor before converting to rust - and in c the constants were defined using `#define` and not `const int` or so)